src/components/AChat/AChat.vue

```diff
@@ -216,25 +198,68 @@ export default {
           block: 'center'
         })
       })
-    },
+    }
+
+    const isScrolledToBottom = () => {
+      if (!messagesRef.value) return false
 
-    isScrolledToBottom() {
       const scrollOffset =
-        this.$refs.messages.scrollHeight -
-        Math.ceil(this.$refs.messages.scrollTop) -
-        this.$refs.messages.clientHeight
+        messagesRef.value.scrollHeight -
+        Math.ceil(messagesRef.value.scrollTop) -
+        messagesRef.value.clientHeight
 
       return scrollOffset <= 60
-    },
+    }
 
-    /**
-     * Returns sender address and name.
-     * @param {string} senderId Sender address
-     * @returns {{ id: string, name: string }}
-     */
-    getSenderMeta(senderId) {
-      return this.partners.find((partner) => isStringEqualCI(partner.id, senderId))
+    const getSenderMeta = (senderId: string) => {
+      return props.partners.find((partner) => isStringEqualCI(partner.id, senderId))
+    }
+
+    onMounted(() => {
+      attachScrollListener()
+
+      if (messagesRef.value) {
+        currentClientHeight.value = messagesRef.value.clientHeight
+        const resizeHandler = () => {
+          if (!messagesRef.value) return
+
+          const clientHeightDelta = currentClientHeight.value - messagesRef.value.clientHeight
+
+          const nonVisibleClientHeight =
+            messagesRef.value.scrollHeight -
+            messagesRef.value.clientHeight -
+            Math.ceil(messagesRef.value.scrollTop)
+          const scrolledToBottom = nonVisibleClientHeight === 0
+
+          if (!scrolledToBottom) {
+            messagesRef.value.scrollTop += clientHeightDelta
+          }
+
+          currentClientHeight.value = messagesRef.value.clientHeight
+        }
```

Comment:
Can we move and define `resizeObserver` outside `onMounted` method? Will it work?

Reply:
I mean `resizeHandler`, not resizeObserver